frontend/src/components/dashboard/dashboard-details.tsx

```diff
@@ -0,0 +1,367 @@
+import React from "react";
+import { connect } from "react-redux";
+import { withStyles } from "tss-react/mui";
+// Components
+import { Container, Grid, Paper } from "@mui/material";
+// Tables
+import ClusterDashboardTableStyled from "./clusters-dashboard-table";
+import AgentDashboardTableStyled from "./agents-dashboard-table";
+import EntriesDashBoardTableStyled from "./entries-dashboard-table";
+import DashboardDrawerStyled from "./dashboard-drawer";
+import { RootState } from 'redux/reducers';
+import { Theme } from '@mui/material/styles'; 
+const styles = (theme:Theme)  => ({
+  root: { 
+    marginTop: -25,
+    marginLeft: -20,
+    display: "flex",
+  },
+  appBarSpacer: theme.mixins.toolbar,
+  content: {
+    flexGrow: 1,
+  },
+  container: {
+    //container for root
+    paddingTop: theme.spacing(4),
+    paddingBottom: theme.spacing(4),
+    marginLeft: 0,
+  },
+  paper: {
+    //container for all grids
+    padding: theme.spacing(2),
+    display: "flex",
+    overflow: "auto",
+    flexDirection: "column",
+    marginBottom: 20,
+  },
+  detailsTitle: {
+    fontSize: 20,
+    color: "rgb(89, 103, 185)",
+  },
+  metadataTag: {
+    fontWeight: "bold",
+    fontSize: 17,
+    marginTop: 10,
+  },
+
+  dashboardDetalsLine: {
+    marginLeft: 0,
+    width: 1200,
+  },
+  metadataDetails: {
+    fontSize: 17,
+  },
+});
+
+// DashboardDetails takes in the selected data and props from dashboard table for a specific entity
+// returns details page for dashboard
+
+type DashboardDetailsProps = {
+  // the clicked dashboard table
+  globalClickedDashboardTable: string,
+  selectedData: {
+    name: string;
+    parentId: string;
+    clusterName: string;
+    spiffeid: string;
+    length: number;
+    numNodes: number;
+    numEntries:number;
+    id:string;
+    created:string;
+    status:string;
+    platformType:string;
+    adminFlag:boolean;
+    entryExpireTime:string;
+    
+  },
+  selectedDataKey:any[],
+  
+
+
+
+} 
+
+  // getClusterMetadata(entry: ClustersList, globalEntries: EntriesList[], globalAgents: AgentsList[]) {
+  //   return {
+  //     id: entry.name,
+  //     name: entry.name,
+  //     created: entry.creationTime,
+  //     numNodes: entry.agentsList.length,
+  //     numEntries: this.numberEntriesOfCluster(entry, globalEntries, globalAgents),
+  //   }
+  // }
```

Comment:
Seems to be a lot of extra space here

```suggestion
} 
```